packages/resources/src/RemixSite.ts

```diff
@@ -0,0 +1,1535 @@
+    ]);
+  }
+
+  private createServerFunctionRoleForEdge(): iam.Role {
+    return this.createServerFunctionRole([
+      "lambda.amazonaws.com",
+      "edgelambda.amazonaws.com",
+    ]);
+  }
+
+  private createServerFunctionRole(principals: string[]): iam.Role {
+    const { defaults } = this.props;
+
+    // Create function role
+    const role = new iam.Role(this, `ServerLambdaRole`, {
+      assumedBy: new iam.CompositePrincipal(
+        ...principals.map((principal) => new iam.ServicePrincipal(principal))
+      ),
+      managedPolicies: [
+        iam.ManagedPolicy.fromManagedPolicyArn(
+          this,
+          "EdgeLambdaPolicy",
+          "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+        ),
+      ],
+    });
+
+    // Attach permission
+    this.cdk.bucket.grantReadWrite(role);
+    if (defaults?.function?.permissions) {
+      attachPermissionsToRole(role, defaults.function.permissions);
+    }
+
+    return role;
+  }
+
+  private createServerFunctionForRegional(bundlePath: string): lambda.Function {
+    const { defaults, environment } = this.props;
+
+    return new lambda.Function(this, `ServerFunction`, {
+      description: "Server handler for Remix",
+      handler: "server.handler",
+      currentVersionOptions: {
+        removalPolicy: RemovalPolicy.DESTROY,
+      },
+      logRetention: logs.RetentionDays.THREE_DAYS,
+      code: lambda.Code.fromAsset(bundlePath),
+      runtime: lambda.Runtime.NODEJS_16_X,
+      memorySize: defaults?.function?.memorySize || 512,
+      timeout: Duration.seconds(defaults?.function?.timeout || 10),
+      role: this.serverLambdaRole,
+      environment,
+    });
+  }
+
+  private createServerFunctionForEdge(bundlePath: string): lambda.IVersion {
+    const name = "Server";
+
+    // Create function asset
+    const asset = new s3Assets.Asset(this, `ServerFunctionAsset`, {
+      path: bundlePath,
+    });
+
+    // Create function based on region
+    const root = this.node.root as App;
+    return root.region === "us-east-1"
+      ? this.createServerFunctionInUE1(name, asset, bundlePath)
+      : this.createServerFunctionInNonUE1(name, asset, bundlePath);
+  }
+
+  private createServerFunctionInUE1(
+    name: string,
+    asset: s3Assets.Asset,
+    assetPath: string
+  ): lambda.IVersion {
+    const { defaults } = this.props;
+
+    // Create function
+    const fn = new lambda.Function(this, `${name}Function`, {
+      description: `${name} handler for Remix`,
+      handler: "server.handler",
+      currentVersionOptions: {
+        removalPolicy: RemovalPolicy.DESTROY,
+      },
+      logRetention: logs.RetentionDays.THREE_DAYS,
+      code: lambda.Code.fromAsset(assetPath),
+      runtime: lambda.Runtime.NODEJS_16_X,
+      memorySize: defaults?.function?.memorySize || 512,
+      timeout: Duration.seconds(defaults?.function?.timeout || 10),
+      role: this.serverLambdaRole,
+    });
+
+    // Create alias
+    fn.addAlias("live");
```

Comment:
**question:** Can we remove this? What is the intention of this? It caused my synth/deploy to fail as there was a conflict.